### PR TITLE
Add perl 5.004_05 (5.4.50).

### PR DESCRIPTION
**Pull request by stikonas**
Enabled bit operations in `gawk`. They are used in `opcode.awk` in the next stage to recreate `opcode.h` file.

Adds perl 5.000 with some `.pl` scripts reimplemented in `awk` for bootstrapping purposes.

Build perl 5.003. This time we can use `perl` from the previous stage to recreate `embed.h`, `keywords.h`, and `opcode.h`. Note that I undefined `__DATE__`, so that final executable is reproducible.

@pder there is unfortunately somewhat silly problem with perl that we will have to solve. Due to float problem in `tcc`, `perl` reports its version to be `0.000` which breaks some things that check for minimal version of `perl`.

Build perl 5.004_50.

### CODE REVIEW

File: sysa/perl5.000/perl5.000.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    rm perly.c perly.h
+    bison -d perly.y
+    mv perly.tab.c perly.c
+    mv perly.tab.h perly.h
+
+    rm embed.h
+    ./embed_h.SH
+
+    rm keywords.h
+    sed -e '1,/__END__/ d' keywords.pl | sed '1d' | awk '{print "#define", "KEY_"$0, NR-1}' > keywords.h
```
Comment (pder) [inline sysa/perl5.000/perl5.000.sh:17]: Could this one-liner be moved into a script named keywords.sh so it is consistent with opcode.sh and embed_h.SH?

**Review by pder — APPROVED** [review]

File: sysa/perl5.000/files/keywords.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh -e
+# SPDX-FileCopyrightText: 2021 Andrius Štikonas <andrius@stikonas.eu>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (fosslinux) [inline sysa/perl5.000/files/keywords.sh:4]: Comment here to the effect of replaces keywords.pl?

File: sysa/perl5.000/files/opcode.sh [hunk 3]
```diff
@@ -0,0 +1,71 @@
+#!/bin/sh -e
+# SPDX-FileCopyrightText: 2021 Andrius Štikonas <andrius@stikonas.eu>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
```
Comment (fosslinux) [inline sysa/perl5.000/files/opcode.sh:5]: Ditto (regarding replaces opcode.pl)

File: sysa/perl5.000/perl5.000.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# SPDX-FileCopyrightText: 2021 Andrius Štikonas <andrius@stikonas.eu>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+src_prepare() {
+    default_src_prepare
+
+    rm perly.c perly.h
```
Comment (fosslinux) [inline sysa/perl5.000/perl5.000.sh:8]: Comment remove pregenerated files or something like that

File: sysa/perl5.000/perl5.000.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    bison -d perly.y
+    mv perly.tab.c perly.c
+    mv perly.tab.h perly.h
+
+    rm embed.h
```
Comment (fosslinux) [inline sysa/perl5.000/perl5.000.sh:13]: Comment Regenerate embed.h, etc

File: sysa/perl5.003/perl5.003.sh [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# SPDX-FileCopyrightText: 2021 Andrius Štikonas <andrius@stikonas.eu>
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+src_prepare() {
+    default_src_prepare
+
+    rm perly.c perly.h
```
Comment (fosslinux) [inline sysa/perl5.003/perl5.003.sh:8]: Comments like from 5.000 please

**Review by fosslinux — CHANGES_REQUESTED** [review]
Looking pretty good, thank you!

Doesn't miniperlmain need to be "generated" using writemain.SH?

File: rootfs.sh [hunk 7]
```diff
@@ -190,6 +190,13 @@ get_file https://ftp.gnu.org/gnu/diffutils/diffutils-2.7.tar.gz
 # gawk 3.0.4
 get_file https://ftp.gnu.org/gnu/gawk/gawk-3.0.4.tar.gz
 
+# perl 5.000
+get_file http://mirrors.develooper.com/perl/really-ancient-perls/oldperl/dist/leo/src/perl5/perl5.000.tar.gz
+
+# perl 5.003
+# Upstream tarball is a bit broken, slackware repackaged it
+get_file https://mirrors.slackware.com/slackware/slackware-3.1/source/d/perl-5.003/perl5.003.tar.gz
+
```
[the remark this answers is not in the document]
Comment (bittorf) [reply rootfs.sh:199]: (and https://github.com/Perl/perl5/archive/perl-5.003.tar.gz)

File: README.md [hunk 8]
```diff
@@ -334,4 +334,28 @@ is required later for autotools.
...
+At this first step we build `miniperl` which is `perl` without support for
+loading modules.
+
+#### Part 30: perl 5.003
+
+We now use `perl` from the previous stage to recreate pre-generated files that
+are shipped in perl 5.003. But for now we still need to use handwritten makefile
+instead of `./Configure` script.
+
+### Part 31: perl 5.004_05
```
Comment (fosslinux) [inline README.md:360]: Newline (sorry nitpicking :P)

**Review by fosslinux — APPROVED** [review]

File: README.md (hunk 8 above)
Comment (fosslinux) [reply README.md:360]: I'll merge this now and you can fix this in your next PR :)

**Pull request merged**